sherpa/astro/tests/test_cache.py

```diff
@@ -32,217 +31,206 @@
+
+    x = numpy.linspace(-5., 5., 100)
+    g1, g2 = Gauss1D(), Gauss1D()
+    g1.fwhm = 1.14
+    g1.pos = 1.2
+    g2.fwhm = 4.13
+    g2.pos = -1.3
+
+    numpy.random.seed(0)
+    y1 = g1(x) + numpy.random.normal(0.0, 0.05, x.shape)
+    y2 = g2(x) + numpy.random.normal(0.0, 0.05, x.shape)
+
+    d4 = Data1D('4', x, y1)
+    d5 = Data1D('5', x, y2)
+
+    return d4, d5
+
+
+def compare_results(expected, got, tol=1e-4):
+
+    for key in ["numpoints", "dof"]:
+        assert int(getattr(got, key)) == expected[key]
+
+    for key in ["istatval", "statval"]:
+        assert float(getattr(got, key)) == pytest.approx(expected[key], rel=tol)
+
+    assert got.parvals == pytest.approx(expected['parvals'], rel=tol)
+
+
+def test_diff_cache(setUp3):
+    d1, d2, d3 = setUp3
+
+    poly1 = Polynom1D()
+    poly2 = Polynom1D()
+    poly3 = Polynom1D()
+    poly1.pars[1].thaw()
+    poly2.pars[1].thaw()
+    poly3.pars[1].thaw()
+    sdata = DataSimulFit('d123', (d1, d2, d3))
+    smodel = SimulFitModel('diff', (poly1, poly2, poly3))
+    sfit = Fit(sdata, smodel, method=NelderMead(), stat=Cash())
+    result = sfit.fit()
+    compare_results(fit_diff_poly_bench, result)
+
+
+def test_same_cache(setUp3):
+    d1, d2, d3 = setUp3
+
+    poly = Polynom1D()
+    poly.pars[1].thaw()
+    sdata = DataSimulFit('d1d2d3', (d1, d2, d3))
+    smodel = SimulFitModel('same', (poly, poly, poly))
+    sfit = Fit(sdata, smodel, method=NelderMead(), stat=Cash())
+    result = sfit.fit()
+    compare_results(fit_same_poly_bench, result)
+
+
+def test_gauss_gauss(setUp2):
+    d4, d5 = setUp2
+
+    g1, g2 = Gauss1D(), Gauss1D()
+    g1.fwhm = 1.3
+    g1.pos = 1.5
+    g2.fwhm = 4.
+    g2.pos = -2.0
+    sdata = DataSimulFit('d4d5', (d4, d5))
+    smodel = SimulFitModel('g1g2', (g1, g2))
+    sfit = Fit(sdata, smodel, method=LevMar(), stat=LeastSq())
+    result = sfit.fit()
+    compare_results(fit_g2g2_bench, result)
+
+
+def test_cache_copy(clean_astro_ui):
+
+    # fake up a PHA data set
+    chans = numpy.arange(1, 11, dtype=numpy.int8)
+    counts = numpy.ones(chans.size)
+
+    # bin width is not 0.1 but something slightly different
+    ebins = numpy.linspace(0.1, 1.2, num=chans.size + 1)
+    elo = ebins[:-1]
+    ehi = ebins[1:]
+
+    dset = ui.DataPHA('test', chans, counts)
+
+    # make sure ARF isn't 1 to make sure it's being applied
+    arf = ui.create_arf(elo, ehi, specresp=0.7 * numpy.ones(chans.size))
+
+    rmf = ui.create_rmf(elo, ehi, e_min=elo, e_max=ehi)
+
+    ui.set_data(1, dset)
+    ui.set_arf(1, arf)
+    ui.set_rmf(1, rmf)
+
+    ui.set_source(ui.const1d.mdl)
+
+    # again not 1
+    mdl.c0 = 8
+
+    # Copy the values from the plot structures, since get_xxx_plot
+    # returns the same object so m1.y == m2.y will not note a difference.
+    #
+    d1y = ui.get_data_plot().y.copy()
+    m1y = ui.get_model_plot().y.copy()
+    s1y = ui.get_source_plot().y.copy()
+
+    d2y = ui.get_data_plot().y.copy()
```

Comment:
I don't understand what this is testing? How could this possibly have a different value than `d1y`? I don't see anything happening in between that would modify the value or the cache. But I also see that it's not new here, it's just taken from the old test, so maybe I don't need to understand.

Reply:
As far as I can tell, it tests that computers are deterministic. If you ask for the same answer twice, you get the same answer...

Reply:
I can't remember the details, but calls to get_model_plot/get_source_plot could change the data structure that get_data_plot uses, so this is likely a check that we aren't overwriting things we shouldn't be. Or at least that is a possible interpretation..